### PR TITLE
[ExportVerilog] Move local name legalization to pre-pass

### PR DESCRIPTION
**Pull request by uenoku**
This PR removes ModuleNameManager from ExportVerilog. Previously it was necessary for ExportVerilog to manage local names and perform legalization online since ExportVerilog had to spill temporary wires while emission for syntax legalization. However, we separated all spilling logics into Prepare, it's possible to know all operations which can declare names. 

* GlobalNameResolver was changed to legalize all the names parallelly. 
* Remove ModuleNameManager from ExportVerilog and look up names through `getSymOpName` and `getVerilogValueName` functions. 
* Prepare was modified to return `LogicalResult` so that we can detect unlowered operations more early (this could be merged in a separated commit).

Basically there is no change in output verilog except for a tiny change of port and parameter names. This is caused by the change of the order of name legalization. (we legalized ports first, but this PR changes to legalize module parameters first).

### CODE REVIEW

File: test/Conversion/ExportVerilog/name-legalize.mlir [hunk 1]
```diff
@@ -11,9 +11,9 @@ hw.module @namechange(%casex: i4) -> (if: i4) {
 hw.module.extern @module_with_bool<bparam: i1>() -> ()
 
 // CHECK-LABEL: module parametersNameConflict
-// CHECK-NEXT:    #(parameter [41:0] p1_0 = 42'd17,
+// CHECK-NEXT:    #(parameter [41:0] p1 = 42'd17,
 // CHECK-NEXT:      parameter [0:0]  wire_0) (
-// CHECK-NEXT:    input [7:0] p1);
+// CHECK-NEXT:    input [7:0] p1_0);
 hw.module @parametersNameConflict<p1: i42 = 17, wire: i1>(%p1: i8) {
   %myWire = sv.wire : !hw.inout<i1>
```
Comment (seldridge) [inline test/Conversion/ExportVerilog/name-legalize.mlir:18]: Can you instead rewrite the test to not have a conflict? We may want to error/warn on port/param collisions in the future.

**Review by seldridge — APPROVED** [review]
I glanced through this quickly and this seems like an excellent cleanup!

I'm excited for how this will make things like https://github.com/llvm/circt/issues/4561.
